packages/optimizer/lib/DomTransformer.js

```diff
@@ -17,6 +17,50 @@
 
 const path = require('path');
 const treeParser = require('./TreeParser.js');
+const log = require('./log.js');
+
+/**
+ * AMP Optimizer Configuration only applying AMP validity perserving transformations.
+ */
+const TRANSFORMATIONS_VALID_AMP = [
+  // Optimizes script import order
+  // needs to run after ServerSideRendering
+  'ReorderHeadTransformer',
+  // needs to run after ReorderHeadTransformer
+  'RewriteAmpUrls',
+  'GoogleFontsPreconnect',
+  'PruneDuplicateResourceHints',
+];
+
+/**
+ * AMP Optimizer Configuration applying all available AMP optimizations including Server-Side_Rendering.
+ */
+const TRANSFORMATIONS_ALL = [
+  // Adds a link to the valid AMP version
+  'AddAmpLink',
+  // Applies server-side-rendering optimizations
+  'ServerSideRendering',
+  // Removes ⚡ or 'amp' from the html tag
+  'RemoveAmpAttribute',
+  // Removes the boilerplate
+  // needs to run after ServerSideRendering
+  'AmpBoilerplateTransformer',
+  // Optimizes script import order
+  // needs to run after ServerSideRendering
+  'ReorderHeadTransformer',
+  // needs to run after ReorderHeadTransformer
+  'RewriteAmpUrls',
+  'GoogleFontsPreconnect',
+  'PruneDuplicateResourceHints',
+  'AddBlurryImagePlaceholders',
+];
+
+const DEFAULT_CONFIG = {
+  verbose: false,
+  validAmp: false,
```

Comment:
It seems that both `validAmp` and `transformers` control which transformers will run? I'm wondering if it be more clear to provide factory methods to create the transformers. eg:

`DomTransformer.allTransformations(verbose: false)` and `DomTransformar.validAmp(verbose: false)`

Those would create a config object and call the constructor with the value for verbose and the correct list of transformers.

Reply:
I'm thinking that it probably would be best to get away from users having to handle transformers on their own. I think it'd be better to handle everything via configuration:

```
const optimiser = AmpOptimizer.create({
  validAmp: true,
  verboseTrue: true,
  preProcessors: [ ... list of transformers ... ],
  postProcessors: [ ... list of transformers ... ]
});
```

Reply:
This PR works towards this, but doesn't break existing APIs yet.

Reply:
We're going to have more PRs moving in that direction then. Sounds Good.